plugins/kubernetes/ingress_kong.go

```
@@ -0,0 +1,265 @@
+	if err != nil {
+		return nil, err
+	}
+	input.ClientCertificate = clientCertificate.String()
+
+	clientKey, err := e.GetArtifact("client_key")
+	if err != nil {
+		return nil, err
+	}
+	input.ClientKey = clientKey.String()
+
+	certificateAuthority, err := e.GetArtifact("certificate_authority")
+	if err != nil {
+		return nil, err
+	}
+	input.CertificateAuthority = certificateAuthority.String()
+
+	serviceType, err := e.GetArtifact("type")
+	if err != nil {
+		return nil, err
+	}
+	input.Type = serviceType.String()
+
+	service, err := parseService(e)
+	if err != nil {
+		return nil, err
+	}
+
+	input.Service = service
+
+	if serviceType.String() == "loadbalancer" {
+
+		apexDomain, err := e.GetArtifact("controlled_apex_domain")
+		if err != nil {
+			return nil, err
+		}
+		input.ControlledApexDomain = apexDomain.String()
+
+		upstreamRoutes, err := e.GetArtifact("upstream_routes")
+		if err != nil {
+			return nil, err
+		}
+
+		input.UpstreamRoutes = parseUpstreamRoutes(upstreamRoutes)
+
+		selectedIngress, err := e.GetArtifact("ingress")
+		if err != nil {
+			return nil, err
+		}
+
+		ingressControllers, err := e.GetArtifact("ingress_controllers")
+		if err != nil {
+			return nil, err
+		}
+
+		// Guarantee persisted ingress controller is configured on the extension side.
+		found := false
+		for _, controller := range strings.Split(ingressControllers.String(), ",") {
+			if controller == selectedIngress.String() {
+				found = true
+			}
+			continue
+		}
+		if found == false {
+			return nil, fmt.Errorf("Selected Ingress Controller is Not Configured")
+
+		}
+
+		parsedController, err := parseController(selectedIngress.String())
+		if err != nil {
+			return nil, err
+		}
+		input.Controller = *parsedController
+
+	}
+
+	return &input, nil
+
+}
+
+func parseUpstreamRoutes(a transistor.Artifact) []UpstreamRoute {
+	var upstreamRoutes []UpstreamRoute
+	for _, upstream := range a.Value.([]interface{}) {
```

Can we check that the type assertions is valid beforehand? Otherwise, we'll get a silent failure and execution will also stop. You can do so like this:
```
upstreamRoutes, ok := a.Value.([]interface{})
if !ok {
  log.Debug(...)
}
```

More info can be found here: https://stackoverflow.com/questions/37609749/how-to-log-type-assertion-errors-in-golang